global-privacy-control/README.md

```diff
@@ -0,0 +1,57 @@
+# Feature Name Tests
+
+Privacy Feature: https://app.asana.com/0/1198207348643509/1199115248606508/f
+
+## Goals
+
+This set of tests verifies implementation of Global Privacy Control signal. In particular it focuses on verifying that:
+
+- that the GPC header appears on all requests
+- that `navigator.globalPrivacyControl` API is available in all frames
+- that the right thing happens when user opts out of GPC and
+- that excluded domains, as defined in the privacy remote configuration, are taken into account.
+
+## Structure
+
+Test suite specific fields:
+
+- `siteURL` - string - currently loaded website's URL (as seen in the URL bar)
+- `frameURL` - string - URL of an iframe in which the feature is operating (optional - if not set assume main frame context)
+- `requestType` - mostly "image" or "main_frame" (navigational request), but can be any of https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/ResourceType - type of the resource being fetched
+- `gpcUserSettingOn` - boolean - if user controlled GPC setting is on or off (optional - if not set assume on)
+- `expectGPCAPI` - boolean - if we expect GPC API to be available in given conditions
+- `expectGPCAPIValue` - "true", "false" or "undefined" - stringified expected value of `Navigator.prototype.globalPrivacyControl`
```

Comment:
values are stringified mostly because `undefined` is not a valid JSON. I could use `null` instead, but that's not the same and would require awkward conversion when integrating tests

Reply:
Is "undefined" consistently the stringified output on all platforms? Seems like it might be a JS-centric convention?  In Swift for example, there is no such thing, only `nil`.  

Maybe sticking with JSON convention and letting each platform do the appropriate JSON-native mapping might make more sense?

Reply:
GPC API is a browser API, it’s only interacted with via JS. We expect JS “undefined” in some of the tests and I think putting there null would be confusing. I can make that clearer in the description.